src/app/public/src/modules/help-key/help-key.component.ts

```diff
@@ -1,20 +1,25 @@
-import { Component, Input, OnInit, OnDestroy } from '@angular/core';
+import { Component, Input, OnDestroy } from '@angular/core';
 
 import { HelpWidgetService } from '../shared';
 @Component({
   selector: 'bb-help-key',
   template: ''
```

Comment:
I feel like we had this discussion already, but have you thought about making this a directive instead? A component is literally just a directive without a template (and since this doesn't have a template, it fits the definition of a directive). I guess you want the convenience of a separate tag? I'm assuming most folks using the Help Widget will be using Stache to some degree, so being able to add the `helpKey` attribute to the `stache` tag would have some benefits.

```
<stache helpKey="default.html">
```

@blackbaud-brandonhare Do you have an opinion on this?

Reply:
Yeah it was discussed a while back, I was thinking the same thing.  However, after talking with Paul and Bobby, even though it doesn't have a template it was a lot clearer and safer to make it a component.  It's easier to look for a specific element on your page every time you go to make an update, than a directive that can live anywhere.  Having it live on another component or html element means it's subject to accidentally get removed or altered through some side affect or just without realizing what it is.   As the directive would not have any bearing on those elements it also seemed strange.  By keeping it a component it won't affect the view, but is self contained and easy to identify and adjust later.

Reply:
Fair enough! Thanks for the explanation.